hcatalog/server-extensions/src/main/java/org/apache/hive/hcatalog/listener/DbNotificationListener.java

```diff
@@ -174,12 +174,17 @@ public class DbNotificationListener extends TransactionalMetaStoreEventListener
 
   //cleaner is a static object, use static synchronized to make sure its thread-safe
   private static synchronized void init(Configuration conf) throws MetaException {
-    if (cleaner == null) {
+    long freq = MetastoreConf.getTimeVar(conf, MetastoreConf.ConfVars.EVENT_DB_LISTENER_CLEAN_INTERVAL, TimeUnit.MILLISECONDS);
+    if (cleaner == null && freq > 0) {
       cleaner =
```

Comment:
The `cleaner` object is accessed in many places in this class and in order to ensure that we are not going to have a NPE we need to review all calls carefully. For instance, it seems that configuration changes (`onConfigChange`) may touch try to update some aspects of the `cleaner` assuming that the latter one is initialized.